benchmark/Streamly/Benchmark/Prelude/Serial.hs

```diff
@@ -459,6 +519,26 @@ minimumBy = S.minimumBy compare
 maximumBy :: Monad m => SerialT m Int -> m (Maybe Int)
 maximumBy = S.maximumBy compare
 
+{-# INLINE the #-}
+the :: Monad m => SerialT m Int -> m (Maybe Int)
+the = S.the
+
+{-# INLINE drainN #-}
+drainN :: Monad m => Int -> SerialT m Int -> m ()
+drainN val = S.drainN val
+
+{-# INLINE drainWhile #-}
+drainWhile :: Monad m => SerialT m Int -> m ()
+drainWhile = S.drainWhile (const True)
+
+{-# INLINE (!!) #-}
+(!!) :: Monad m => Int -> SerialT m Int -> m (Maybe Int)
+(!!) val = flip (Internal.!!) val
+
+{-# INLINE lookup #-}
+lookup :: Monad m => Int -> SerialT m Int -> m (Maybe Int)
+lookup val = S.lookup val . S.map (\x -> (x, x))
```

Comment:
Depending on the val we are either finding the key early in the stream or never. Better to split it into two benchmarks `lookupFirst` `lookupNever`. Similarly we can also have 2 benchmarks for `the` as well along the same lines.